test/lz4_compression.jl

```diff
@@ -0,0 +1,149 @@
+@testset "lz4_compression" begin
+    text = b"""
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean sollicitudin
+    mauris non nisi consectetur, a dapibus urna pretium. Vestibulum non posuere
+    erat. Donec luctus a turpis eget aliquet. Cras tristique iaculis ex, eu
+    malesuada sem interdum sed. Vestibulum ante ipsum primis in faucibus orci luctus
+    et ultrices posuere cubilia Curae; Etiam volutpat, risus nec gravida ultricies,
+    erat ex bibendum ipsum, sed varius ipsum ipsum vitae dui.
+    """
+
+    @testset "Transcoding" begin
+
+        @test LZ4FastCompressorStream <: TranscodingStream
+        @test LZ4SafeDecompressorStream <: TranscodingStream
+
+        compressed = transcode(LZ4FastCompressor, text)
+        @test sizeof(compressed) < sizeof(text)
+
+        decompressed = transcode(LZ4SafeDecompressor, compressed)
+        @test sizeof(decompressed) > sizeof(compressed)
+        @test decompressed == Vector{UInt8}(text)
+
+        test_roundtrip_fileio(LZ4FastCompressor, LZ4SafeDecompressor)
+        test_roundtrip_transcode(LZ4FastCompressor, LZ4SafeDecompressor)
+
+        file = IOBuffer(text)
+        stream = LZ4SafeDecompressorStream(LZ4FastCompressorStream(file))
+        flush(stream)
+
+        @test read(stream) == text
+        close(stream)
+        close(file)
+
+        file = IOBuffer(text)
+        stream = LZ4SafeDecompressorStream(LZ4FastCompressorStream(file; acceleration = 5))
+        flush(stream)
+
+        @test read(stream) == text
+        close(stream)
+        close(file)
+
+        file = IOBuffer("")
+        stream = LZ4SafeDecompressorStream(LZ4FastCompressorStream(file))
+        flush(stream)
+
+        @test read(stream) == b""
+        close(stream)
+        close(file)
+
+        teststring = rand(UInt8, 10000)
+        file = IOBuffer(teststring)
+        stream = LZ4SafeDecompressorStream(LZ4FastCompressorStream(file; block_size = 2048); block_size = 2048)
+        flush(stream)
+
+        @test read(stream) == teststring
+        close(stream)
+        close(file)
+
+    end
+
+    @testset "Errors" begin
+        # Malformed decompression input
```

Comment:
What are your thoughts on creating a `testset` for these types of errors? Having specific testsets it's easier to debug failures in unit testing I've found?